Gems/Atom/RHI/Vulkan/Code/Source/RHI/ShaderResourceGroupPool.cpp

```diff
@@ -34,15 +34,21 @@ namespace AZ
 
             m_descriptorSetCount = RHI::Limits::Device::FrameCountMax;
 
+#if !defined(AZ_PLATFORM_ANDROID)
             const uint32_t constantSize = layout.GetConstantDataSize();
             if (constantSize > 0)
             {
                 BufferPoolDescriptor bufferPoolDescriptor;
                 bufferPoolDescriptor.m_bindFlags = RHI::BufferBindFlags::Constant;
                 bufferPoolDescriptor.m_heapMemoryLevel = RHI::HeapMemoryLevel::Host;
                 m_constantBufferPool = BufferPool::Create();
+                static int index = 0;
+                m_constantBufferPool->SetName(Name(AZStd::string::format("SRG_%s_%d", layout.GetName().GetCStr(), ++index)));
                 m_constantBufferPool->Init(device, bufferPoolDescriptor);
             }
+#else
```

Comment:
Platform specific code can not go in like this. You have to sue the PAL layer. Take a look at C:\GitHub\o3de\Gems\Atom\RHI\Vulkan\Code\Include\Platform to get better idea. Also I think we can just adopt a single allocator for all platforms. Dont need any sort of platform specific code for SRG constants.

Reply:
done.
`AZ_PLATFORM_ANDROID` has been removed.